gsoc-2020/extension-activity-monitor.md

```diff
@@ -0,0 +1,49 @@
+# Extension Activity Monitor 
+
+**Mentor:** Luca Greco, ...
+
+**Email:** lgreco@mozilla.com
+
+## Project Description
+
+Extensions do most of their work invisibly from users, and so  Extension activity is a complete mystery for most users, even advanced ones.
+
+Providing more transparency could help increase reliability of abuse reporting and accountability for developers, as well as providing an
+additional useful tool to aid investigating bugs in the extensions or in the Firefox WebExtensions internals.
+
+## Skills Required
+
+An applicant needs familiarity with:
+
+* HTML/JavaScript and the Web API
+* Revision Control Systems (e.g. git and/or mercurial)
+
+and the following skills:
+
+* Contributing to Open Source
+* Think critically
+* Good reading comprehension
+* Do independent research
+* Enjoy working with others
+* Be productive with little supervision
+* Able to work remotely
+
+## Project Details
+
+An `activityLog` API is available (in Firefox >= 70) to privileged extensions, the applicant goal is leveraging this API to
+create a privileged extension which would explore various ways to present this information to the users.
+
+In particular, this project goal is to expore the following areas:
+
+- Browser tab (about:extensionmonitor, maybe) that shows all extensions active in all open tabs, extensions altering search, and for each installed extension
+  show activity like network requests and data interactions between content scripts and background scripts.
```

Comment:
Can we use the activity log to monitor network requests? That's neat! Maybe this could be in a devtools panel to save some design/frontend efforts.

Reply:
I've based the draft on the description we had in the related card from our trello board, and this part was already mentioned in the original description.

From a technical perspective, being able to monitor network requests originated from an extension should be part of [Bug 1558366 - add dom activity support for activityLog](https://bugzilla.mozilla.org/show_bug.cgi?id=1558366), which has a patch but it has not been reviewed nor landed yet.

I've not removed from the project draft because I thought that it may still be possible that we could actually complete and land it before the GSOC project would be starting. 
@mixedpuppy what do you think about this?